packages/react/src/componentsV2/swapsForm/index.tsx

```diff
@@ -0,0 +1,185 @@
+  formValues,
+  setFieldValue,
+  normalizedPrice,
+}: SwapsFormProps) => {
+  const {
+    paymentDetails,
+    tokenSwapLoading,
+    tokenSwapCurrencies,
+    getTokenSwapCurrencies,
+    tokenSwapQuote,
+    getTokenSwapQuote,
+    tokenSwapError,
+    currencyList,
+  } = useHelioProvider();
+
+  const { HelioSDK } = useCompositionRoot();
+
+  const [selectedCurrency, setSelectedCurrency] = useState(
+    tokenSwapCurrencies?.find(
+      ({ symbol }) => symbol === formValues.swapsCurrency
+    )
+  );
+
+  const debouncedQuantity = useDebounce(formValues.quantity, DEBOUNCE_TIME);
+  const debouncedPrice = useDebounce(formValues.price, DEBOUNCE_TIME);
+
+  useEffect(() => {
+    getTokenSwapCurrencies();
+  }, [paymentDetails?.currency?.mintAddress]);
+
+  useEffect(() => {
+    const currency = currencyList?.find(
+      (it) => formValues.currency === it.symbol
+    );
+
+    if (selectedCurrency?.mintAddress != null) {
+      getTokenSwapQuote(
+        paymentDetails.id,
+        'PAYLINK' as PaymentRequestType.PAYLINK,
+        selectedCurrency.mintAddress,
+        formValues.quantity,
+        normalizedPrice,
+        paymentDetails?.dynamic ? currency?.mintAddress : undefined
+      );
+    }
+  }, [
+    selectedCurrency,
+    debouncedQuantity,
+    debouncedPrice,
+    paymentDetails.currency.mintAddress,
+    paymentDetails.currency.symbol,
+    paymentDetails.id,
+    paymentDetails.type,
+  ]);
+
+  const currencyOptions =
+    tokenSwapCurrencies
+      ?.filter(
+        ({ mintAddress, symbol }) =>
+          mintAddress != null && symbol !== paymentDetails?.currency.symbol
+      )
+      .map((currency) => ({
+        value: currency.symbol,
+        label: `${currency.symbol}`,
+        icon: <CurrencyIcon gradient iconName={currency.symbol} />,
+      })) || [];
+
+  const getCurrencyLabel = (): string => {
+    let currencyLabel = selectedCurrency
+      ? `${selectedCurrency?.symbol}`
+      : 'Select currency';
+
+    if (tokenSwapQuote != null && selectedCurrency != null && !tokenSwapError) {
+      currencyLabel = `${HelioSDK.tokenConversionService.convertFromMinimalAndRound(
+        selectedCurrency?.symbol,
+        tokenSwapQuote?.inAmount
+      )} ${selectedCurrency?.symbol}`;
+    }
+
+    return currencyLabel;
+  };
+
+  return (
+    <StyledSwapsContainer>
+      {tokenSwapLoading ? (
+        <StyledSpinner>
+          <Spinner />
+        </StyledSpinner>
+      ) : (
+        <div>
+          <div>
+            <SelectBox
+              label="Paying in"
+              options={currencyOptions}
+              value={getCurrencyLabel()}
+              fieldName="swapsCurrency"
+              showValidations
+              onChange={(option) => {
+                setFieldValue('swapsCurrency', option.value);
+                setSelectedCurrency(
+                  HelioSDK.currencyService.getCurrencyBySymbol(
+                    String(option.value)
+                  ) as unknown as Currency
```

Comment:
Do we really need the as uknown as Currency? does getCurrencyBySymbol not return the correct type? If you are certain it returns currency you could also use !